### PR TITLE
BSP-356: fixed all issues for FormA

### PR DESCRIPTION
**Pull request by yarpo**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BSP-356

### Change description ###
Update all that should be updated in FormA

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java [hunk 1]
```diff
@@ -60,10 +60,10 @@
         commaSeparatedEntryTransformer(OcrFieldName.DISCHARGE_PERIODICAL_PAYMENT_SUBSTITUTE, "dischargePeriodicalPaymentSubstituteFor",
                 dischargePeriodicalPaymentSubstituteChecklistToCcdFieldNames, ocrDataFields, transformedCaseData);
 
-        AddressMapper.applyMappings("applicantSolicitor", "solicitorAddress", ocrDataFields, transformedCaseData);
-        applyMappingsForAddress("applicant", ocrDataFields, transformedCaseData);
-        applyMappingsForAddress("respondent", ocrDataFields, transformedCaseData);
-        AddressMapper.applyMappings("respondentSolicitor", "rSolicitorAddress", ocrDataFields, transformedCaseData);
+        applyMappingsForAddressExtended("applicantSolicitor", "solicitorAddress", ocrDataFields, transformedCaseData);
+        applyMappingsForAddressExtended("applicant","applicantAddress", ocrDataFields, transformedCaseData);
```
Comment (jacmurphy) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java:64]: Please add a space after the comma

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java [hunk 2]
```diff
@@ -191,12 +191,35 @@ private void commaSeparatedEntryTransformer(String ocrNameWithCommaSeparatedValu
...
+        exceptionRecordToCcdFieldsMap.put(OcrFieldName.RESPONDENT_SOLICITOR_FIRM, "rSolicitorFirm");
+        exceptionRecordToCcdFieldsMap.put(OcrFieldName.RESPONDENT_SOLICITOR_NAME, "rSolicitorName");
+
         return exceptionRecordToCcdFieldsMap;
     }
 
-    private void applyMappingsForAddress(
-            String prefix, List<OcrDataField> ocrDataFields, Map<String, Object> modifiedMap) {
-        applyMappings(prefix, prefix + "Address", ocrDataFields, modifiedMap);
+    private void applyMappingsForAddressExtended(
```
Comment (jacmurphy) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java:200]: Would `applyMappingsForAddress` ever be used after this is implemented ?  If not, i would think that refactoring it to account for your changes would be beneficial

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java [hunk 3]
```diff
@@ -191,12 +191,35 @@ private void commaSeparatedEntryTransformer(String ocrNameWithCommaSeparatedValu
...
+
+        addMappingsTo(
+            parentField,
+            extendedAddress(nestedFieldPrefix),
+            modifiedMap,
+            ocrDataFields
+        );
+    }
+
+    private static ImmutableMap<String, String> extendedAddress(String prefix) {
```
Comment (jvictorcf) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java:212]: I'd call this something like "getExtendedAddressMapForPrefix"

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java (hunk 2 above)
Comment (jvictorcf) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java:200]: I'd call this applyMappingsForExtendedAddress

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java [hunk 4]
```diff
@@ -191,12 +191,35 @@ private void commaSeparatedEntryTransformer(String ocrNameWithCommaSeparatedValu
...
+
         return exceptionRecordToCcdFieldsMap;
     }
 
-    private void applyMappingsForAddress(
-            String prefix, List<OcrDataField> ocrDataFields, Map<String, Object> modifiedMap) {
-        applyMappings(prefix, prefix + "Address", ocrDataFields, modifiedMap);
+    private void applyMappingsForAddressExtended(
+            String prefix, String parentField, List<OcrDataField> ocrDataFields, Map<String, Object> modifiedMap) {
+        String nestedFieldPrefix = StringUtils.capitalize(prefix + "Address");
```
Comment (jvictorcf) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java:202]: Maybe you can put this line inside the "extendedAddress" method.

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulk/scan/validation/FormAValidatorTest.java [hunk 5]
```diff
@@ -220,7 +219,7 @@ public void shouldFailFieldsHavingInvalidValues() {
 
         assertThat(validationResult.getStatus(), is(WARNINGS));
         assertThat(validationResult.getErrors(), is(emptyList()));
-        assertThat(validationResult.getWarnings().size(), is(24));
+        assertThat(validationResult.getWarnings().size(), is(23));
```
Comment (jvictorcf) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulk/scan/validation/FormAValidatorTest.java:222]: I question the utility of this line.
Happy to keep, but if you agree this doesn't add value, I'm happy to support its removal.

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java (hunk 2 above)
Comment (yarpo) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java:200]: It's temporary. If it all works well I'd move it to bsp-common-lib. There is no place now where we use address without address line 2.

**Comment by vianvi** [comment]
`src/functionalTests/resources/json/bulkscan/formA.json` - Hasn't updated with new fields

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulk/scan/validation/FormAValidatorTest.java (hunk 5 above)
Comment (yarpo) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulk/scan/validation/FormAValidatorTest.java:222]: well, it caught the fact I have changed logic, so it actually adds little value imho :)

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java (hunk 2 above)
Comment (yarpo) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java]: nope, it's all moved to bsp lib now.

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java (hunk 3 above)
Comment (yarpo) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java]: removed. All in bsp-common-lib now

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java (hunk 2 above)
Comment (yarpo) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java]: refactored

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java [hunk 6]
```diff
@@ -65,8 +64,8 @@
                 dischargePeriodicalPaymentSubstituteChecklistToCcdFieldNames, ocrDataFields, transformedCaseData);
 
         AddressMapper.applyMappings("applicantSolicitor", "solicitorAddress", ocrDataFields, transformedCaseData);
-        applyMappingsForAddress("applicant", ocrDataFields, transformedCaseData);
-        applyMappingsForAddress("respondent", ocrDataFields, transformedCaseData);
+        AddressMapper.applyMappings("applicant", "applicantAddress", ocrDataFields, transformedCaseData);
```
Comment (jvictorcf) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java:67]: static import?

Comment (jacmurphy) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/bulkscan/transformation/FormAToCaseTransformer.java:67]: Do we have Transformer tests? - @yarpo

**Pull request merged**